benchmarks/definition/core.py

```diff
@@ -0,0 +1,65 @@
+"""Definition of the Benchmark class."""
+
+import time
+from collections.abc import Callable
+from datetime import datetime
+from typing import Generic, TypeVar
+from uuid import UUID
+
+from attrs import define, field
+from attrs.validators import instance_of, is_callable
+from pandas import DataFrame
+
+from benchmarks.result.metadata_class import ResultMetadata
+from benchmarks.result.result import Result
+
+BenchmarkConfig = TypeVar("BenchmarkConfig")
+BenchmarkFunction = Callable[[BenchmarkConfig], DataFrame]
+
+
+@define
+class Benchmark(Generic[BenchmarkConfig]):
+    """A class to define a benchmark task."""
+
+    name: str = field(validator=instance_of(str))
+    """The name of the benchmark."""
+
+    settings: BenchmarkConfig = field()
+    """The configuration for the benchmark settings."""
+
+    benchmark_callable: BenchmarkFunction = field(validator=is_callable())
+    """The function that executes the benchmark code and returns
+    the results as well as metadata."""
+
+    identifier: UUID = field(validator=instance_of(UUID))
+    """The unique identifier of the benchmark running which can be set
+    to compare different executions of the same benchmark setting."""
+
+    @property
+    def description(self) -> str:
+        """The description of the benchmark callable."""
+        if self.benchmark_callable.__doc__ is None:
+            return ""
+        return self.benchmark_callable.__doc__
+
+    def run(self) -> Result:
```

Comment:
I just realized: why not simply rebrand this as `__call__`? That's the perfect use case for it, no?